lib/application.js

```diff
@@ -124,6 +124,9 @@ app.model = function(Model, config) {
   if (arguments.length > 1) {
     config = config || {};
     configureModel(Model, config, this);
+    this.on('modelRemoted', function() {
+      setSharedMethodSharedProperties(Model, this, config);
```

Comment:
So the problem was that `setSharedMethodSharedProperties` wasn't exported anywhere. This made it difficult to call it from somewhere useful and setup only one watcher. I've moved it to its own file so it can be called from `loopback.js` and `application.js`. Let me know if you can think of a better solution.